### PR TITLE
feat: Rework

### PR DESCRIPTION
**Pull request by mxschmitt**
Changes:
- updated `date-fns` to 2.0.0
- added auto-deployment of GitHub Pages
- updated example to latest create-react-app version

TODO:
- Some maintainer has to add a GitHub API Token to Travis: (https://travis-ci.org/Hacker0x01/react-datepicker/settings env: `GITHUB_TOKEN`)

### CODE REVIEW

**Comment by mxschmitt** [comment]
> Not sure if we're able to test without actually doing this, but I like the changes and it makes things easier for us too. Let's give it a shot!

what exactly, the auto deployment? I've tested it in my fork and it works: https://travis-ci.org/mxschmitt/react-datepicker/builds/574762063#L547
🚀

**Comment by martijnrusschen** [comment]
Just to confirm, what kind of API token do we need here? Is this the personal access token?

**Comment by mxschmitt** [comment]
> Just to confirm, what kind of API token do we need here? Is this the personal access token?

yep, personal Access Token with the scopes described here: https://docs.travis-ci.com/user/deployment/pages/#setting-the-github-token

**Comment by martijnrusschen** [comment]
Merged, watching the build logs to see if this is working as expected.

**Comment by martijnrusschen** [comment]
All working, thanks!

File: scripts/release.sh [hunk 1]
```diff
@@ -7,8 +7,8 @@ git checkout .
 
 npm version $1
 
-npm install -g react-docgen
-react-docgen ./src/*.jsx | ./scripts/buildDocs.sh
+yarn global add react-docgen
+e
```
Comment (martijnrusschen) [inline scripts/release.sh]: Hi @mxschmitt I think this is breaking how we generate our docs. It seems we don't run the react-docgen tool anymore to build docs and push this.

**Pull request merged**